mycore-mods/src/main/java/org/mycore/mods/classification/MCRTypeOfResource.java

```diff
@@ -39,33 +39,49 @@ class MCRTypeOfResource extends MCRAuthorityInfo {
     /**
      * The mods:typeOfResource code, which is same as the category ID
      */
-    private String code;
+    private final String code;
 
     MCRTypeOfResource(String code) {
         this.code = code;
     }
 
+    /**
+     * @deprecated Use {@link #parseXML(Element)} instead
+     */
+    @Deprecated
+    public static MCRTypeOfResource getAuthorityInfo(Element modsElement) {
+        return parseXML(modsElement);
+    }
+
     /**
      * If the given element is mods:typeOfResource, returns the MCRTypeOfResource mapping.
      */
-    public static MCRTypeOfResource getAuthorityInfo(org.jdom2.Element modsElement) {
+    public static MCRTypeOfResource parseXML(Element modsElement) {
```

Comment:
the switch of the default import from org.jdom2.Element to w3c.dom.Element is confusing. (The class supports both Elmement objects)

Reply:
I switched the order, (W3C first, then JDom2) so that this class has the same internal structure as `MCRAuthorityAndCode` and `MCRAuthorityWithURI`.